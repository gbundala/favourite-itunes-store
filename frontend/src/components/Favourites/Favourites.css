.favourite-items-wrapper {
  display: flex;
  flex-wrap: wrap;
  gap: 16px;
}
