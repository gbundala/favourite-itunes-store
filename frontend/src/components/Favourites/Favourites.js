// Import react
import React from "react";

// Import child component
import ItunesItem from "../ItunesItem/ItunesItem";

// Import stylesheet
import "./Favourites.css";

export default function Favourites({ favouriteItunes, setFavouriteItunes }) {
  // Handler to remove and item from favourites
  // we call the filter method on the array of favourites
  // and return elements that are not equal to the
  // id of the item that has been selected
  function handleRemoveFavouriteItem(uniqueItunesItemId) {
    setFavouriteItunes(
      favouriteItunes.filter(
        (favouriteItem) => favouriteItem.uniqueId !== uniqueItunesItemId
      )
    );
  }

  return (
    <div>
      <h4>Favourite iTunes</h4>
      <div className="favourite-items-wrapper">
        {favouriteItunes &&
          favouriteItunes.map((itunesItem) => {
            return (
              <ItunesItem
                key={itunesItem.uniqueId}
                itunesItem={itunesItem}
                handleRemoveFavouriteItem={handleRemoveFavouriteItem}
              />
            );
          })}
      </div>
    </div>
  );
}
