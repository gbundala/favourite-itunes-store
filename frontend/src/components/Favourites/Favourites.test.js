// Import libaries and frameworks
import React from "react";
import testRenderer from "react-test-renderer";

// Import component to be tested
import Favourites from "./Favourites";

// Snapshot testing
test("Favourites component renders inline with snapshot", () => {
  const tree = testRenderer.create(<Favourites />).toJSON();

  expect(tree).toMatchSnapshot();
});
