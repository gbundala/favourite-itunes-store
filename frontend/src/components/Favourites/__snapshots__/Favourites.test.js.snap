// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourites component renders inline with snapshot 1`] = `
<div>
  <h4>
    Favourite iTunes
  </h4>
</div>
`;
